src/main/java/help/swgoh/api/SwgohAPI.java

```diff
@@ -111,360 +105,574 @@ public static Optional<Language> fromUnicode( String unicode )
+     *
+     * @param allyCode Ally code of any guild member in guild to request.
+     * @param language Optional language to return translated names. If no language specified, no translations will be applied.
+     * @param filter   Optional projection of response fields you want returned. If no fields are specified, all fields will be returned.
+     * @return guild data by individual ally code of guild member.
+     */
+    CompletableFuture<List<Guild>> getFullLargeGuild(int allyCode, Language language, SwgohAPIFilter filter);
+
+    default CompletableFuture<List<Guild>> getFullLargeGuild(int allyCode, SwgohAPIFilter filter) {
+        return getFullLargeGuild(allyCode, null, filter);
+    }
+
+    default CompletableFuture<List<Guild>> getFullLargeGuild(int allyCode, Language language) {
+        return getFullLargeGuild(allyCode, language, SwgohAPIFilter.ALL);
     }
+
+    default CompletableFuture<List<Guild>> getFullLargeGuild(int allyCode) {
+        return getFullLargeGuild(allyCode, null, SwgohAPIFilter.ALL);
+    }
+
+    /**
+     * Returns the current event schedule.
+     * <p>
+     * Cache sync: 4 hours
+     * <p>
+     * https://api.swgoh.help/swgoh
+     *
+     * @param language Optional language to return translated names. If no language specified, no translations will be applied.
+     * @param filter   Optional projection of response fields you want returned. If no fields are specified, all fields will be returned.
+     * @return current event schedule.
+     */
+    CompletableFuture<Events> getFullEvents(Language language, SwgohAPIFilter filter);
+
+    default CompletableFuture<Events> getFullEvents(Language language) {
+        return getFullEvents(language, SwgohAPIFilter.ALL);
+    }
+
+    default CompletableFuture<Events> getFullEvents(SwgohAPIFilter filter) {
+        return getFullEvents(null, filter);
+    }
+
+    default CompletableFuture<Events> getFullEvents() {
+        return getFullEvents(null, SwgohAPIFilter.ALL);
+    }
+
+    /**
+     * An improved version of the {@link #getRosters(List, Language, SwgohAPIFilter)} method, which instead of a raw JSON as a
+     * {@code java.lang.String} responds with an {@link List} of {@link PlayerRoster}s.
+     * <p>
+     * Cache sync:
+     * Registered user: 4 hours
+     * Verified user: 2 hours
+     * Patreon user: 1 hour
+     * <p>
+     * https://api.swgoh.help/swgoh
+     *
+     * @param allyCodes Ally code(s) of the players to request.
+     * @param language  Optional language to return translated names. If no language specified, no translations will be applied.
+     * @param filter    Optional projection of response fields you want returned. If no fields are specified, all fields will be returned.
+     * @return an object representation of the players rosters
+     */
+    List<PlayerRoster> getImprovedRosters(List<Integer> allyCodes, Language language, SwgohAPIFilter filter) throws ExecutionException, InterruptedException;
```

Comment:
Rather than "getImproved-", these methods should also follow the same naming convention as the others, whether that is "getFull-" or something else.

Reply:
I see now that the implementation for this method calls `/players` in the background instead of `/rosters`. Is there a reason why you chose that? As it is now, this is basically just a helper method to get the roster data from the `/players` endpoint. But now there is no typed alternative for the `/rosters` endpoint response. We should at least provide a method for each endpoint first, and then possibly add helper methods.

Reply:
The '''/roster''' endpoint is really annoying to deal with since instead of an "toon"-object for every toon with an attribute name it has a field with the name of the character and the data for that character stored in the object. For example the will be a field 
'''
"magmaTrooper"={
"level"=85,
...
}
'''

That makes it really hard to deserialize the JSON into an object and the class would have to be extended every time a new character or ship is released. There probably is a solution to this problem but at the moment I got no idea how to solve it.

Reply:
For the helper method: I'm fine with removing or renaming them, but as I said in the comment before using the ```/roster``` endpoint is really annoying.

Reply:
Changed the method name to ```getPlayerRosterObject``` and the javadoc to explain what the method is actually doing. Still got to figure out a way to deal with the response of the ```/roster``` endpoint but I will do at a later time.